airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/blob_transfer.py

```diff
@@ -41,19 +24,12 @@ def write_streams(
         stream_reader: AbstractFileBasedStreamReader,
         logger: logging.Logger,
     ) -> Iterable[Dict[str, Any]]:
-        file_no = 0
         try:
-            data_generator = self._file_reader.read_data(config, file, stream_reader, logger, self.file_read_mode)
-            local_writer = LocalFileTransferClient()
-            for file_opened, file_size in data_generator:
-                yield local_writer.write(file.uri, file_opened, file_size, logger)
-                file_no += 1
+            yield stream_reader.get_file(file=file, local_directory=self._local_directory, logger=logger)
```

Comment:
Implementation sample for source-s3 [here](https://github.com/airbytehq/airbyte/pull/48346/files#diff-b5e8d7cf7e79880dd2e77408410825335670decbc45f4960d2cf07ac323bae86).

Reply:
Manual testing in [this workspace](https://cloud.airbyte.com/workspaces/166f3e04-9021-4f63-a963-a43abb548cec/connections/d7e111cf-1ddc-41bc-8b1a-34bf435c3f2f/timeline) seems be working fine for sync > 5GB :)